src/sources/hydra.rs

```diff
@@ -143,11 +144,43 @@ pub struct Stage {
 
 pub struct Worker {
     socket: HydraConnection,
-    intersect: Point,
+    intersect: WorkerIntersect,
+}
+
+/// Worker state for finding the right intersection point
+#[derive(Debug, Clone)]
+pub enum WorkerIntersect {
+    SkipUntil(u64, Vec<u8>), // Possibility of Point::Origin is excluded
+    ProcessMessages,
 }
 
 impl Worker {
-    async fn process_next(
+    async fn process(&mut self, stage: &mut Stage, msg: HydraMessage) -> Result<(), WorkerError> {
+        let point = msg.pseudo_point();
+        match &self.intersect {
+            WorkerIntersect::SkipUntil(slot, hash) => {
+                let target = Point::Specific(slot.clone(), hash.clone());
+                debug!(
+                    "Skipping message {} before or at intersection {}",
```

Comment:
```suggestion
                    "Skipping message {} as it is before the requested intersection {}",
```

Reply:
maybe this is better wording ...

Reply:
The reason for the awkward `or at` wording is that we can have `Skipping message 10 before or at intersection 10`... Could have a dedicated different log message for that though, like some `Skipping final message 10 at intersection 10`, at the cost of a special case

Reply:
Or maybe `Skipping message 10 as it is before (or at) the requested intersection 10`, for the sake of having the `0 ... 9` cases signal that message 10 also will be skipped.